src/lib/list/selection-list.ts

```diff
@@ -444,11 +443,12 @@ export class MatSelectionList extends _MatSelectionListMixinBase implements Focu
     const optionIndex = this._getOptionIndex(option);
 
     if (optionIndex > -1 && this._keyManager.activeItemIndex === optionIndex) {
+      const options = this.options.toArray();
       // Check whether the option is the last item
       if (optionIndex > 0) {
-        this._keyManager.updateActiveItemIndex(optionIndex - 1);
-      } else if (optionIndex === 0 && this.options.length > 1) {
-        this._keyManager.updateActiveItemIndex(Math.min(optionIndex + 1, this.options.length - 1));
+        this._keyManager.updateActiveItem(options[optionIndex - 1]);
```

Comment:
You can pass in a number to `updateActiveItem` to avoid both the deprecated method and the call to `toArray` above.

Reply:
👍